### ISSUE TITLE
Add Statistical Window Functions

### ISSUE THREAD

**Issue by rupurt** [cube2222/octosql#306]
It would be awesome to have statistical window functions like with the `OVER` clause like Postgres & DuckDB. Is this something on the radar?

**Comment by cube2222** [comment]
Yes! Actually, something I'm planning to add whenever I have some time to spend on OctoSQL. I really want these too.
